### PR TITLE
[YQL-17709] [channel] split spilling into actor and non-actor parts

### PR DESCRIPTION
**Pull request by lll-phill-lll**
### Changelog entry 

Removed direct calls of actors. Splitted channel spiller into actor and non-actor parts and replaced calls with events sending.

### Changelog category 

* Improvement

### Additional information

Arcadia PR with tests: https://a.yandex-team.ru/review/5620461/details

...

### CODE REVIEW

**Review by Darych — DISMISSED** [review]

File: ydb/library/yql/dq/actors/spilling/channel_storage.cpp [hunk 1]
```diff
@@ -20,62 +20,109 @@ using namespace NActors;
...
+        blob = std::move(it->second.ExtractValue());
+        LoadingBlobs_.erase(it);
+        --StoredBlobsCount_;
+
+        return true;
+    }
+
+private:
+    void UpdateWriteStatus() {
+        TVector<THashMap<ui64, TWritingBlobInfo>::iterator> itersToRemove;
```
Comment (resetius) [inline ydb/library/yql/dq/actors/spilling/channel_storage.cpp:95]: useless var

File: ydb/library/yql/dq/actors/spilling/channel_storage.cpp [hunk 2]
```diff
@@ -20,62 +20,109 @@ using namespace NActors;
...
+        for (auto it = WritingBlobs_.begin(); it != WritingBlobs_.end();) {
```
Comment (resetius) [inline ydb/library/yql/dq/actors/spilling/channel_storage.cpp:96]: use erase_if: https://en.cppreference.com/w/cpp/container/unordered_map/erase_if

Comment (lll-phill-lll) [reply ydb/library/yql/dq/actors/spilling/channel_storage.cpp:96]: But this is not std::unordered_map, I used THashMap. And THashMap doesn't have erase_if method. Do you suggest to replace it with std::unordered_map?

File: ydb/library/yql/dq/actors/spilling/channel_storage.cpp (hunk 1 above)
Comment (lll-phill-lll) [reply ydb/library/yql/dq/actors/spilling/channel_storage.cpp]: thanks!

File: ydb/library/yql/dq/actors/spilling/channel_storage.cpp (hunk 2 above)
Comment (lll-phill-lll) [reply ydb/library/yql/dq/actors/spilling/channel_storage.cpp:95]: fixed

**Review by resetius — APPROVED** [review]

**Pull request merged**